tests/infra_test.go

```diff
@@ -693,12 +696,21 @@ var _ = Describe("[Serial]Infrastructure", func() {
 			}
 
 			errorCount := 0
+			unknownErrorTypeFound := false
 			for ix := 0; ix < concurrency; ix++ {
 				err := <-errors
 				if err != nil {
-					errorCount += 1
+					var e *neturl.Error
+					if errs.As(err, &e) {
+						Expect(e.Err.Error()).To(ContainSubstring("Client.Timeout exceeded while awaiting headers"))
```

Comment:
You have not answered or resolved the issue with the text that was pointed out by @RamLavi in previous comments.
You should not assert against text from libs which you do not control. Especially when you have an alternative.

Reply:
As i wrote, see please `IsProbableEOF` vendor/k8s.io/apimachinery/pkg/util/net/http.go
The same method is used there.
Also what is the alternative you suggest ? 
The only thing that is return beside this string is timeout = true in that struct, and it might be too generic to use it.

func IsProbableEOF(err error) bool {
	if err == nil {
		return false
	}
	if uerr, ok := err.(*url.Error); ok {
		err = uerr.Err
	}
	msg := err.Error()
	switch {
	case err == io.EOF:
		return true
	case msg == "http: can't write HTTP request on broken connection":
		return true
	case strings.Contains(msg, "http2: server sent GOAWAY and closed the connection"):
		return true
	case strings.Contains(msg, "connection reset by peer"):
		return true
	case strings.Contains(strings.ToLower(msg), "use of closed network connection"):
		return true
	}
	return false
}

Reply:
btw the error comes from `/usr/local/go/src/net/http/client.go`
```
				err = &httpError{
					// TODO: early in cycle: s/Client.Timeout exceeded/timeout or context cancelation/
					err:     err.Error() + " (Client.Timeout exceeded while awaiting headers)",
					timeout: true,
				}
```
I believe, if it helps

Reply:
I think the timeout check will suffice.
Referencing me to something that is not a timeout is not relevant and that code is unsafe due to the text check. It may worth the risk there, but here I will take the risk of the timeout check.

Reply:
will make sure it catch the ipv6 error and use it if so